src/source/ObjcGenerator.scala

```diff
@@ -178,7 +178,7 @@ class ObjcGenerator(spec: Spec) extends Generator(spec) {
     val self = marshal.typename(objcName, r)
 
     refs.header.add("#import <Foundation/Foundation.h>")
-    refs.body.add("!#import " + q(spec.objcIncludePrefix + (if (r.ext.objc) "../" else "") + marshal.headerName(ident)))
+    refs.body.add("!#import " + q(spec.objcIncludePrefix + (if (r.ext.objc) spec.objcExtendedRecordIncludePrefix else "") + marshal.headerName(ident)))
```

Comment:
Those are two different things. `--objc-include-prefix` specifies the include path Djinni has to generate in order for the compiler to find the Objective-C files *it is generating*. `--objc-extended-record-include-prefix` specifies what include path Djinni has to generate for the compiler to find *your own handwritten code*. Those paths don't have to be related at all.